### ISSUE TITLE
A few issues

### ISSUE THREAD

**Issue by dehlirious** [farisc0de/Uploady#17]
Non-docker! Just copied the PHP files and went.

If you're on signup.php , and you click the "TOS" (should be ToS I believe) -> it redirects back to login.php, making the ToS unreadable.
(I know "The page you are looking for does not exist. " even when you're able to access the ToS but still, just saying) 

On signout, `PHP message: PHP Warning:  Undefined array key "redirect" in Uploady/logout.php on line 9`

Also "Could not instantiate mail function. " on signup (i don't have emailing setup) but the point is, that is Not something that should be public facing. That's something that should be shown to an admin, not a member of the public trying to sign up. No logged error occurred so I would not have known of the issue if I didn't test it 

Then after not verifying email and signing in, trying to upload an image, error.
```
PHP Warning:  fopen(/REDACTED.jpg): 
Failed to open stream: Permission denied in Uploady/vendor/farisc0de/phpfileuploading/src/Upload.php on line 748
PHP Fatal error:  Uncaught TypeError: stream_set_timeout(): 
Argument #1 ($stream) must be of type resource, false given in Uploady/vendor/farisc0de/phpfileuploading/src/Upload.php:751
```

fopen is trying to upload to the root directory .. ? /img.jpg , an absolute path, instead of the relative Uploady directory, or more specifically inside the uploads folder?

Line 748: `$fp = fopen($this->upload_folder['folder_path'] . "/" . $filename, 'w');`
configuration: `define("UPLOAD_FOLDER", "uploads");`
Maybe my configuration is to blame because it isnt /var/www/blah/blah/uploads , but still the Uploads.php file acted like this variable `$this->upload_folder['folder_path']` was completely empty 

Thanks for your time and effort! I'm really enjoying what you've created, if I have time I may try and contribute.

**Comment by farisc0de** [comment]
Thank you so much for reporting this, for the first part I fixed the issue in the last commit, the terms page was not whitelisted.

I am looking into the second part as it is a weird behavior and should not happen.

**Comment by dehlirious** [comment]
> Thank you so much for reporting this, for the first part I fixed the issue in the last commit, the terms page was not whitelisted.
> 
> I am looking into the second part as it is a weird behavior and should not happen.

Awesome! I'll have to try it out again, and yeah I may be able to spend some time to find out why as well but recently my heads been in another project

**Comment by farisc0de** [comment]
For the second part can try to deploy using the steps in this video, and let me know what happens:

https://youtu.be/NTtFekrDna4

**Comment by dehlirious** [comment]
Yeah, this issue is fixed.

Sidenote:
I tried to install Uploady, using the latest version on github and ran into this error(on install.php) when improper mysql user/password was set I ran into this error

`PHP Fatal error:  Uncaught Error: Call to a member function prepare() on null in Uploady/src/Uploady/Database.php:121`

Is this a big deal? No, not really. But a check , to return a error message instead of a PHP fatal error is ideal! E.g. when a bad username/password , tell us as such "Bad user/pass!" or "No database exists!" etc etc!
